.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs

```diff
@@ -189,6 +190,32 @@ private List<Token> SingleTimePointWithPatterns(string text, DateObject referenc
                     string beforeString = text.Substring(0, (int)extractionResult.Start);
                     ret.AddRange(GetTokenForRegexMatching(beforeString, config.WeekOfRegex, extractionResult));
                     ret.AddRange(GetTokenForRegexMatching(beforeString, config.MonthOfRegex, extractionResult));
+
+                    // Cases like "2 days from/before today/tomorrow"
+                    var isAgo = this.config.AgoRegex.Match(extractionResult.Text).Success;
+                    var isLater = this.config.LaterRegex.Match(extractionResult.Text).Success;
+
+                    if (isAgo || isLater)
+                    {
+                        ret.AddRange(GetTokenForRegexMatching(beforeString, config.LessThanRegex, extractionResult));
+                        ret.AddRange(GetTokenForRegexMatching(beforeString, config.MoreThanRegex, extractionResult));
+
+                        // cases like "within 3 days from yesterday/tomorrow" does not make any sense
```

Comment:
Improve comment to describe the two cases for 'within'

Reply:
Fixed.